### PR TITLE
[818] Support of Occurence export.

### PR DESCRIPTION
**Pull request by ebausson-obeo**
# PLEASE READ ALL ITEMS AND CHECK ONLY RELEVANT CHECKBOXES BELOW

## Project management

- [X] Has the pull request been added to the relevant milestone?
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [X] Have the relevant issues been added to the pull request?
- [X] Have the relevant labels been added to the issues? (`area:`, `type:`)
- [X] Have the relevant issues been added to the same project milestone as the pull request?

## Changelog and release notes

- [X] Has the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] In case of a change with a visual impact, are there any screenshots in the `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] In case of a key change, has the change been added to `Key highlights` section in `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc` + `doc/content/modules/user-manual/pages/release-notes/YYYY.MM.0.adoc`?

## Documentation

- [ ] Have you included an update of the [documentation](https://doc.mbse-syson.org) in your pull request? Please ask yourself if an update (installation manual, user manual, developer manual...) is needed and add one accordingly.

## Tests

- [X] Is the code properly tested? Any pull request (fix, enhancement or new feature) should come with a test (or several). It could be unit tests, integration tests or cypress tests depending on the context. Only doc and releng pull request do not need for tests.

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -84,7 +84,7 @@ This reduces the time required to build SysON, especially when using Spring Tool
 - https://github.com/eclipse-syson/syson/issues/805[#805] [import] Improve non regression tests.
 Added source sysml file and unaltered ast.json result.
 - https://github.com/eclipse-syson/syson/issues/812[#812] [export] Fix visibility issue when resolving name of privately imported element during export.
-
+- https://github.com/eclipse-syson/syson/issues/818[#818] [export] Add partial support of OccurrenceDefinition to export.
 === New features
```
[the remark this answers is not in the document]
Comment (ebausson-obeo) [reply CHANGELOG.adoc:89]: done.

File: CHANGELOG.adoc [hunk 2]
```diff
@@ -84,7 +84,7 @@ This reduces the time required to build SysON, especially when using Spring Tool
 - https://github.com/eclipse-syson/syson/issues/805[#805] [import] Improve non regression tests.
 Added source sysml file and unaltered ast.json result.
 - https://github.com/eclipse-syson/syson/issues/812[#812] [export] Fix visibility issue when resolving name of privately imported element during export.
-
+- https://github.com/eclipse-syson/syson/issues/818[#818] [export] Add partial support of OccurrenceDefinition to export.
```
[the remark this answers is not in the document]
Comment (AxelRICHARD) [reply CHANGELOG.adoc]: No it's not

[the remark this answers is not in the document]
Comment (ebausson-obeo) [reply CHANGELOG.adoc]: My bad, now it's done.

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 3]
```diff
@@ -254,6 +258,19 @@ public String casePartUsage(PartUsage partUsage) {
         return this.appendDefaultUsage(this.newAppender(), partUsage).toString();
     }
 
+    @Override
+    public String caseOccurrenceUsage(OccurrenceUsage occurrenceUsage) {
+        String selectedKeyword;
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:263]: The name of the variable seems not to reflect its usage. Please update it.

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 4]
```diff
@@ -475,10 +492,14 @@ public String caseLiteralInteger(LiteralInteger literal) {
     }
 
     private String appendDefaultUsage(Appender builder, Usage usage) {
+        return appendKeywordUsage(builder, usage, this.getUsageKeyword(usage));
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:495]: Missing `this.`

File: backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java [hunk 5]
```diff
@@ -475,10 +492,14 @@ public String caseLiteralInteger(LiteralInteger literal) {
...
+    }
+
+    private String appendKeywordUsage(Appender builder, Usage usage, String keyword) {
```
Comment (AxelRICHARD) [inline backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java:498]: Why do you name this method `appendKeywordUsage`? It does much more than that.

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/OccurrenceUsageImpl.java [hunk 6]
```diff
@@ -71,10 +71,10 @@ public class OccurrenceUsageImpl extends UsageImpl implements OccurrenceUsage {
      * <!-- end-user-doc -->
      *
      * @see #getPortionKind()
-     * @generated
+     * @generated NOT
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/OccurrenceUsageImpl.java:74]: Please add the metamodel changes in a separate issue+PR

**Pull request closed without merging**